components/homme/src/share/namelist_mod.F90

```diff
@@ -582,20 +571,7 @@ subroutine readnl(par)
            test_case(1:13)== "jw_baroclinic"  .or. &
            test_case(1:5) == "dcmip"          .or. &
            test_case(1:5) == "mtest"          .or. &
-           test_case      == "planar_hydro_gravity_wave"            .or. &
```

Comment:
Clean up needless checks. This follows the pattern of the other test sets: check just the prefix.